library/std/src/thread/mod.rs

```diff
@@ -1001,10 +1001,13 @@ impl ThreadId {
         unsafe {
             let _guard = GUARD.lock();
 
-            // If we somehow use up all our bits, panic so that we're not
+            // If we somehow use up all our bits, abort so that we're not
             // covering up subtle bugs of IDs being reused.
             if COUNTER == u64::MAX {
-                panic!("failed to generate unique thread ID: bitspace exhausted");
+                drop(_guard); // in case this ends up calling `ThreadId::new()`, avoid reentrant lock acquire.
+                // Avoid panic machinery.
+                eprintln!("failed to generate unique thread ID: bitspace exhausted");
+                crate::process::abort();
```

Comment:
I believe you can use `rtabort!()`.

Reply:
Unfortunately I can't... that macro is declared in `sys_common`, which must be imported after `thread` since it needs the `thread_local!` macro.

Reply:
Can you turn it into a decl macro 2.0?

Reply:
AFAIK those still do not allow mutually recursive modules on the macro level.

Reply:
Decl macro 2.0 resolve like regular items. This compiled:

```rust
#![feature(decl_macro)]
mod a {
    pub macro a_macro() {}
    crate::b::b_macro!();
}

mod b {
    pub macro b_macro() {}
    crate::a::a_macro!();
}
```